src/extension/sldService/src/test/java/org/geoserver/sldservice/rest/ClassifierTest.java

```diff
@@ -1987,4 +1992,134 @@ private void checkRuleLineSymbolizer(Rule rule, String color) {
+                        + "/sldservice/cite:ClassificationPoints2/"
+                        + getServiceUrl()
+                        + ".xml?"
+                        + "attribute=bar&ramp=red&method=jenks&intervals=2&";
+        Document domJenks = getAsDOM(restPathJenks, 200);
+        print(domJenks);
+        ByteArrayOutputStream baosJenks = new ByteArrayOutputStream();
+        print(domJenks, baosJenks);
+        String resultXmlJenks = baosJenks.toString().replace("\r", "").replace("\n", "");
+        Rule[] rulesJenks =
+                checkSLD(
+                        resultXmlJenks
+                                .replace("<Rules>", sldPrefix)
+                                .replace("</Rules>", sldPostfix));
+        assertTrue(rulesJenks.length == 2);
+        Rule firstJenks = rulesJenks[0];
+        Rule secondJenks = rulesJenks[1];
+        checkNotOverlappingRules(firstJenks, secondJenks);
+    }
+
+    @Test
+    public void testNotOverlappingRulesOpen() throws Exception {
+        final String restPath =
+                RestBaseController.ROOT_PATH
+                        + "/sldservice/cite:ClassificationPoints2/"
+                        + getServiceUrl()
+                        + ".xml?"
+                        + "attribute=bar&ramp=red&method=quantile&intervals=3&open=true";
+        Document dom = getAsDOM(restPath, 200);
+        print(dom);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        print(dom, baos);
+        String resultXml = baos.toString().replace("\r", "").replace("\n", "");
+        Rule[] rules =
+                checkSLD(resultXml.replace("<Rules>", sldPrefix).replace("</Rules>", sldPostfix));
+        assertTrue(rules.length == 3);
+        Rule first = rules[0];
+        Rule second = rules[1];
+        checkNotOverlappingRules(first, second);
+        final String restPathJenks =
+                RestBaseController.ROOT_PATH
+                        + "/sldservice/cite:ClassificationPoints2/"
+                        + getServiceUrl()
+                        + ".xml?"
+                        + "attribute=bar&ramp=red&method=jenks&intervals=3&open=true";
+        Document domJenks = getAsDOM(restPathJenks, 200);
+        print(domJenks);
+        ByteArrayOutputStream baosJenks = new ByteArrayOutputStream();
+        print(domJenks, baosJenks);
+        String resultXmlJenks = baosJenks.toString().replace("\r", "").replace("\n", "");
+        Rule[] rulesJenks =
+                checkSLD(
+                        resultXmlJenks
+                                .replace("<Rules>", sldPrefix)
+                                .replace("</Rules>", sldPostfix));
+        assertTrue(rulesJenks.length == 2);
+        Rule firstJenks = rulesJenks[0];
+        Rule secondJenks = rulesJenks[1];
+        checkNotOverlappingRules(firstJenks, secondJenks);
+    }
+
+    private void checkNotOverlappingRules(Rule first, Rule second) {
```

Comment:
Name suggests a general non overlap test, but in fact it's pretty specific and does not test what happens in the first rule filter.
Suggestion for a different approach:

```
SimpleFeatureType ft = (SimpleFeatureType)= getCatalog().getFeatureTypeByName("ClassificationPoints2").getFeatureType();
SimpleFeature feature = DataUtilities.create(ft, "=1|2.0|POINT(4 2.5)");
assertTrue(first.getFilter().evaluate(feature));
assertFalse(second.getFilter().evaluate(feature));
```

Reply:
Thanks for the suggestion, applied.